src/main/java/com/ldtteam/domumornamentum/block/AbstractBlockSlab.java

```diff
@@ -1,20 +1,175 @@
+            boolean xflag = context.getClickLocation().x - (double)context.getClickedPos().getX() > 0.5D;
+            boolean yflag = context.getClickLocation().y - (double)context.getClickedPos().getY() > 0.5D;
+            boolean zflag = context.getClickLocation().z - (double)context.getClickedPos().getZ() > 0.5D;
+            if (xpos || ypos || zpos){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.UP && !yflag && direction != Direction.DOWN){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.DOWN && yflag && direction != Direction.UP){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.NORTH && !zflag && direction != Direction.SOUTH){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.SOUTH && zflag && direction != Direction.NORTH){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.EAST && xflag && direction != Direction.WEST){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.WEST && !xflag && direction != Direction.EAST){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+        }
+
+        /**if holding the shift key while clicking, set slab FACING (N/E/W/S)
+         * all facing slabs use BOTTOM TYPE
+         * else use the base class default */
+        if (context.isSecondaryUseActive()) {
+            if (context.getNearestLookingDirection() == Direction.UP){
+                return defaultstate.setValue(FACING, Direction.UP);
+            }
+            if (context.getNearestLookingDirection() == Direction.DOWN){
+                return defaultstate.setValue(FACING, Direction.DOWN);
+            }
+           return defaultstate.setValue(FACING, context.getHorizontalDirection().getOpposite());
+        }
+        /**default placing */
+        BlockState blockstate1 = this.defaultBlockState().setValue(TYPE, SlabType.BOTTOM).setValue(FACING, Direction.DOWN).setValue(WATERLOGGED, Boolean.valueOf(fluidstate.getType() == Fluids.WATER));
+        return direction != Direction.DOWN && (direction == Direction.UP || !(context.getClickLocation().y - (double)blockpos.getY() > 0.5D)) ? blockstate1 : blockstate1.setValue(FACING, Direction.UP);
```

Comment:
This seems overly complicated. Cant this be just FACING, direction?

Reply:
136/137 are exactly as vanilla slab, it's checking to see if the other side was clicked in.  Above I was just trying to say get opposite only when not up or down